### ISSUE TITLE
LRU auto-strategy for GC

### ISSUE THREAD

**Issue by dmpetrov** [iterative/dvc#855]
It would be convenient to set up a strategy like LRU instead of removing everything or by time #678.

Use cases:
1. `dvc gc --stragegy LRU 10Gb` - release 10Gb from cache
2. `dvc config Core.LRU_Limit 20Gb && dvc run -d input/ -o output.tsv Rscript ` - it will GC automaticaly as a result of `dvc run` if cache size if more then 20Gb

This should work on a shared cache as well for multiple users.

It looks like this appproach will require a special state file in a cache directory to support LRU strategy and some of the DVC commands will modify the file: `dvc run`, `dvc repro` etc.

**Comment by andrethrill** [comment]
If I may step in, it would be cool to have the option for:

3. `dvc gc --stragegy LRU 10Gb` - release files from cache until it is equal to, or less than 10GB.

And raise an error or a warning if it needs to release files required for current working commit/branch.

**Comment by dmpetrov** [comment]
@andrethrill you are right! It is better to specify a limit for cache, then amount to release.
